### ISSUE TITLE
Segmentation result is spotty?

### ISSUE THREAD

**Issue by lolitsgab** [tensorflow/models#7730]
I am running deeplab on the [DeepFashion2 Dataset](https://github.com/switchablenorms/DeepFashion2) and am encountering problems when visualizing my results with vis.py.

Here is what my model got after training for 10000 steps with default train.py and visualizing with default vis.py:
![non-seg](https://user-images.githubusercontent.com/15098502/67801587-b4d4a400-fa46-11e9-9197-60b2fb5d51c6.png)
![seg](https://user-images.githubusercontent.com/15098502/67801590-b56d3a80-fa46-11e9-8efd-e807f88dafc6.png)

This is what a color indexed image looks like in my dataset (not from same picture as above, as I can't find it (my dataset is 180,000 pictures and deeplab shuffles the dataset):
![000017](https://user-images.githubusercontent.com/15098502/67789170-c0b56b80-fa30-11e9-8f71-8c84ee053eca.png)

I am converting the image masks I generated from 32-depth to 8-depth (color-indexed) images using ImageMagick. The above image is a color indexed image. The non-masked image is regular depth.

These are the flags I am using for training:
```
python3 deeplab/train.py \
  --logtostderr \
  --dataset="PQR" \
  --train_split="train" \
  --model_variant="xception_65" \
  --atrous_rates=6 \
  --atrous_rates=12 \
  --atrous_rates=18 \
  --output_stride=16 \
  --decoder_output_stride=4 \
  --train_crop_size=651 \
  --train_crop_size=500 \
  --train_batch_size=2 \
  --training_number_of_steps="10000" \
  --fine_tune_batch_norm=true \
  --tf_initial_checkpoint="/content/pretrained/deeplabv3_pascal_trainval/model.ckpt" \
  --train_logdir="./deeplab/datasets/ckpts/"\
  --dataset_dir="/root/deeplabvc/models/research/deeplab/datasets/PQR/tfrecord"\
  --initialize_last_layer=False\
  --last_layers_contain_logits_only=True
```
Any possible help?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
